src/lib/Scenes/Home/Components/ForYou/index.tsx

```diff
@@ -1,104 +1,126 @@
-    super(props)
-
-    const dataSource: ListViewDataSource = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 })
-
-    this.state = {
-      isRefreshing: false,
-      modules: [],
-      dataSource,
-    }
+  state = {
+    isRefreshing: false,
+    rowData: [],
   }
 
   componentDidMount() {
-    const rows: DataSourceRow[] = []
-    const artworkModules = this.props.forYou.artwork_modules || []
-    const artistModules = this.props.forYou.artist_modules && this.props.forYou.artist_modules.concat()
-    const fairsModule = this.props.forYou.fairs_module
-    rows.push({ type: "fairs", data: fairsModule })
-    for (let i = 0; i < artworkModules.length; i++) {
-      const artworkModule = artworkModules[i]
-      rows.push({ type: "artwork", data: artworkModule })
-      if ((i + 1) % 2 === 0) {
+    const { forYou } = this.props
+    const rowData = []
+    const artworkModules = forYou.artwork_modules || []
+    const artistModules = forYou.artist_modules && forYou.artist_modules.concat()
+    const fairsModule = forYou.fairs_module
+
+    rowData.push({
+      type: "fairs",
+      data: fairsModule,
+    })
+
+    artworkModules.forEach((artworkModule, index) => {
+      rowData.push({
+        type: "artwork",
+        data: artworkModule,
+      })
+
+      const alternateRow = (index + 1) % 2 === 0
+
+      if (alternateRow) {
         const artistModule = artistModules.shift()
         if (artistModule) {
-          rows.push({ type: "artist", data: artistModule })
+          rowData.push({
+            type: "artist",
+            data: artistModule,
+          })
         }
       }
-    }
+    })
+
     this.setState({
-      dataSource: this.state.dataSource.cloneWithRows(rows),
+      rowData,
     })
   }
 
-  handleRefresh = () => {
-    this.setState({ isRefreshing: true })
-    const stopRefreshing = () => this.setState({ isRefreshing: false })
-
-    Promise.all(
-      this.state.modules.map(module => {
-        return new Promise((resolve, reject) => {
-          module.forceFetch(null, readyState => {
-            if (readyState.error) {
-              reject(readyState.error)
-            } else if (readyState.aborted) {
-              reject()
-            } else if (readyState.done) {
-              resolve()
-            }
-          })
-        })
+  registerRailModule = railModule => {
+    this.railModules = this.railModules.concat([railModule])
+  }
+
+  handleRefresh = async () => {
+    this.setState({
+      isRefreshing: true,
+    })
+
+    try {
+      await Promise.all(this.railModules.map(railModule => railModule.refreshData()))
+
+      this.setState({
+        isRefreshing: false,
+      })
+    } catch (error) {
+      console.error("ForYou/index.tsx", error.message)
+
+      this.setState({
+        error: error.message, // TODO: Display this somehow
       })
-    ).then(stopRefreshing, stopRefreshing)
+    }
+  }
+
+  // TODO: Is this still needed?
+  handleScroll = event => {
```

Comment:
Looked through the project for `currentScrollOffset`. Pretty sure, this can go, it isn't being used anymore